constructor/fcp.py

```diff
@@ -198,51 +198,24 @@ def check_duplicates_files(pc_recs, platform, ignore_duplicate_files=True):
     return total_tarball_size, total_extracted_pkgs_size
 
 
-def _precs_from_environment(environment, download_dir, user_conda):
-    from subprocess import check_output
-
-    # get basic data about the environment's packages
-    list_flag = "--prefix" if isdir(environment) else "--name"
-    json_listing = check_output([user_conda, "list", list_flag, environment, "--json"])
-    listing = json.loads(json_listing)
-    packages = {p["dist_name"]: p for p in listing}
-    # get the package install order and MD5 sums,
-    # creating a tuple of dist_name, URL, MD5, filename (fn)
-    explicit = check_output([user_conda, "list", list_flag, environment,
-                             "--explicit", "--json", "--md5"],
-                            universal_newlines=True)
-    ordering = []
-    for line in explicit.splitlines():
-        if not line or line.startswith("#") or line.startswith("@"):
-            continue
-        url, _, md5 = line.rpartition("#")
-        _, _, fn = url.rpartition("/")
-        if fn.endswith(".tar.bz2"):
-            dist_name = fn[:-8]
-        else:
-            dist_name, _ = splitext(fn)
-        ordering.append((dist_name, url, md5, fn))
-
-    # now, create PackageCacheRecords
-    precs = []
-    for dist_name, url, md5, fn in ordering:
-        package = packages[dist_name]
-        package_tarball_full_path = join(download_dir, fn)
-        extracted_package_dir = join(download_dir, dist_name)
-        if 'platform' in package:
-            package['subdir'] = package['platform']
-            del package['platform']
-        precs.append(PackageCacheRecord(url=url, md5=md5, fn=fn,
-                                        package_tarball_full_path=package_tarball_full_path,
-                                        extracted_package_dir=extracted_package_dir,
-                                        **package))
-    return precs
+def _precs_from_environment(environment, dir_path):
+    if not isdir(environment) and ('/' in environment or '\\' in environment):
+        env2 = join(dir_path, environment)
+        if isdir(env2):
+            environment = env2
+    if isdir(environment):
+        environment = abspath(join(dir_path, expanduser(environment)))
+    else:
+        environment = locate_prefix_by_name(environment)
+    pdata = PrefixData(environment)
```

Comment:
The only drawback I can anticipate is that the user wanted to use their supplied `conda-exe` for this, but it's not like `PrefixData` has changed a lot in the past and it doesn't look like it will do so anytime soon.